setup-go/action.yml

```diff
@@ -1,6 +1,6 @@
 ---
 name: 'Setup Go'
-description: 'Installs Go and restores build/test/module cache.'
+description: 'Installs Go and restores build/test/module cache'
```

Comment:
This is the one where I have the biggest "why" :) Why does one need a custom Go? Is it only to restore cache? Or are there any other reasons? My guess was that there is something wrong with the preinstalled ubuntu-20.04 version... But it would be really nice to have more info to not guess ;)

Reply:
Currently, there are two reasons:
1. GitHub-hosted runners currently have 1.17, and we need 1.18+ for both generics and fuzzing.
2. And we don't want to copy&paste steps that tweak cache, run `go mod tidy`, etc.

Also, before 1.18 was released, this action used to download Go tip from https://github.com/AlekSi/golang-tip

Reply:
Yeah, sounds reasonable to me.